app/src/main/java/com/sun/makeupwindow/data/source/local/dao/FavoriteProductDaoImpl.kt

```diff
@@ -0,0 +1,63 @@
+package com.sun.makeupwindow.data.source.local.dao
+
+import android.content.ContentValues
+import com.sun.makeupwindow.data.model.Product
+import com.sun.makeupwindow.data.source.local.db.AppDatabase
+
+class FavoriteProductDaoImpl private constructor(
+    private val appDatabase: AppDatabase
+) : FavoriteProductDao {
+
+    private val writableDatabase = appDatabase.writableDatabase
+    private val readableDatabase = appDatabase.readableDatabase
+
+    override fun getFavoriteProduct(): List<Product> {
```

Comment:
```suggestion
    override fun getProducts(): List<Product> {
```
Từ favorite lặp lại quá nhiều. nên em có thể bỏ các biến khác. class chứá nó có từ Favorite là đc rồi